Mlir

wsmoses opened the pull request.

ftynse committed:
update and modernize to run with enzyme-jax

* drop manual `__enzyme_autodiff` when unguarded by a flag
* modify thrust API
* update makefiles to compile with enzyme-jax

wsmoses committed:
Fix lbm run

wsmoses commented:
@ftynse

ftynse commented on LBM/Makefile:
```diff
@@ -5,15 +5,16 @@
+LIB_RAISE_PATH ?= $(HOME)/git/Enzyme-JaX/bazel-bin/libRaise.so
 REACTANT_EXTRA_PATH ?= $(HOME)/git/Reactant.jl/deps/ReactantExtra/bazel-bin/
 
+export REACTANT_DEBUG=1
```
Drop debug output when no longer needed? (For some reason cannot suggest changes through github)

wsmoses committed:
don't set cache config

wsmoses committed:
add vars

wsmoses committed:
fix

wsmoses committed:
fix

wsmoses committed:
restore

wsmoses committed:
lulesh runs

wsmoses committed:
Update Makefile

vimarsh6739 committed:
Add a new Makefile for builds with embedded-clang

Gets rid of the -fplugin way of building LBM. We still have to use the
-add-plugin option to initiate the custom frontend action.

vimarsh6739 committed:
add EMBEDDED_CLANG opt

vimarsh6739 committed:
update

vimarsh6739 committed:
replace cc1 load with fplugin

wsmoses committed:
Merge pull request #3 from vimarsh6739/mlir2

Add a new Makefile for builds with embedded-clang

wsmoses commented on LBM/Makefile:
```diff
@@ -91,59 +104,52 @@ endif
-else
-  CFLAGS +=  -mllvm -enzyme-loop-invariant-cache=0
-endif
+# ifeq ($(NEWCACHE),yes)
```
make an extra var WITH_RECTANT, which these ifs are inside of [so then enzyme-llvm still works with the makefile]

wsmoses commented on LBM/Makefile:
```diff
@@ -44,16 +51,22 @@ CFLAGS := -mllvm -max-heap-to-stack-size=1000000 -I $(CUDA_PATH)/include -I .
+  ENZYME_LOAD_FLAGS += -fplugin=$(ENZYME_PATH)
+endif
+
+CFLAGS += -ffast-math --cuda-path=$(CUDA_PATH) -L$(CUDA_PATH)/lib64 --cuda-gpu-arch=sm_$(SM_VERSION)  -std=c++11 $(ENZYME_LOAD_FLAGS) -Xclang -add-plugin -Xclang enzyme -mllvm -raising-plugin-path -mllvm $(LIB_RAISE_PATH) -mllvm -reactant-backend -mllvm cuda -x cuda -fno-exceptions
```
this does not need `-Xclang -add-plugin -Xclang enzyme`

xys-syx committed:
match to CUDA 12+

wsmoses committed:
Merge pull request #6 from xys-syx/LULESH-Update

LULESH guard software atomicAdd<double> for CUDA 12+ toolchains
